### DISCUSSION TITLE
vi mode ctrl-c / cancel behavior

### DISCUSSION THREAD

**Discussion by micimize** [akinomyoga/ble.sh#241] [Q&A] [answered]
Hi – I've just installed ble.sh and have been digging through the wiki some, but can't figure out how to regain typical `ctrl-c` behavior in the line editor.

What I'd like to accomplish is the ability to at any point `ctrl-c` and leave behind the buffer in the scrollback with `^C` or some other indicator that it wasn't run so I can mash `ctrl-c` with great fervor like I used to.

As an aside, thinking about it, this is my muscle memory for "nah forget about this and start over and give me some space to think,"  so I'm curious if others have a different workflow for that process.

Thanks,
Michael

**Comment by akinomyoga** [comment, marked as answer]
Thank you for trying ble.sh. Please use `discard-line`. I'll later add this to the Wiki page and the `blerc` template.

```
# blerc
function blerc/vim-load-hook {
  ble-bind -m vi_imap -f C-c discard-line
  ble-bind -m vi_nmap -f C-c discard-line
}
blehook/eval-after-load keymap_vi blerc/vim-load-hook
```

> As an aside, thinking about it, this is my muscle memory for "nah forget about this and start over and give me some space to think," so I'm curious if others have a different workflow for that process.

For example, `HdL` can be used to clear the content of the current linen in the default configuration of ble.sh. There is no way to go to the next line leaving its content in the default configuration. It should be noted that `gg` and `G` operates on the entire history in the default configuration of ble.sh, but they can also be configured with `ble-bind`.

**Reply by micimize** [reply]
Great – that looks to work! Might look into appending `^C` to the line or something later, but the syntax highlighting for run vs unrun commands serves nicely enough.

**Reply by micimize** [reply]
is there a way to bind two commands? I'd like to reset history when discarding lines so `C-r` search resets.

Tried a few things like this but didn't work: `ble-bind -m vi_imap -f C-c 'discard-line && history-end'`

**Reply by akinomyoga** [reply]
It is written at the end of [README §2.7](https://github.com/akinomyoga/ble.sh#27-key-bindings).

**Reply by micimize** [reply]
Ah perfect! Thanks, missed that
